token/core/identity/kvs/storage.go

```diff
@@ -88,3 +87,56 @@ func (s *IdentityStorage) IdentityExists(identity view.Identity, wID identity.Wa
 func (s *IdentityStorage) walletPrefix(wID identity.WalletID) string {
 	return fmt.Sprintf("%s-%s-%s-%s", s.tmsID.Network, s.tmsID.Channel, s.tmsID.Namespace, wID)
 }
+
+type WalletPathStorage struct {
+	kvs    KVS
+	prefix string
+}
+
+func NewWalletPathStorage(kvs KVS, prefix string) *WalletPathStorage {
+	return &WalletPathStorage{kvs: kvs, prefix: prefix}
+}
+
+func (w *WalletPathStorage) AddWallet(id string, path string) error {
+	k, err := kvs.CreateCompositeKey("token-sdk", []string{"msp", w.prefix, "registeredIdentity", id})
+	if err != nil {
+		return errors.Wrapf(err, "failed to create identity key")
+	}
+	return w.kvs.Put(k, path)
+}
+
+func (w *WalletPathStorage) WalletPaths() (identity.Iterator[identity.WalletPath], error) {
+	it, err := w.kvs.GetByPartialCompositeID("token-sdk", []string{"msp", w.prefix, "registeredIdentity"})
+	if err != nil {
+		return nil, errors.WithMessage(err, "failed to get registered identities from kvs")
+	}
+	return &WalletPathStorageIterator{Iterator: it}, nil
+}
+
+type WalletPathStorageIterator struct {
+	kvs.Iterator
+}
+
+func (w *WalletPathStorageIterator) HasNext() bool {
```

Comment:
You can remove `HasNext()` and `Close()`. They are "inherited" from `kvs.Iterator`.